src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -441,8 +454,11 @@ public boolean generateStructureOracleFile(URL solutionRepoURL, URL exerciseRepo
         Repository exerciseRepository = gitService.getOrCheckoutRepository(exerciseRepoURL);
         Repository testRepository = gitService.getOrCheckoutRepository(testRepoURL);
 
+        gitService.resetToOriginMaster(solutionRepository);
```

Comment:
For all 3 repositories we should do a hard reset (in case of e.g. a force push from local), otherwise the endpoint will fail and there will be no way to fix this issue from the client.
I think this approach is ok because it is an action only available to instructors and the GUI shows a warning.